### PR TITLE
LVM-477: Update tutor/student/matches/enrollments/volunteer_jobs

### PR DESCRIPTION
**Pull request by zack**
https://cs5500-jira.ccs.neu.edu/browse/LVM-477

* Volunteer Jobs and Matches associate with affiliates instead of coordinators
* Matches validate on tutor and student having the same affiliate
  through their currently active enrollment or volunteer job

### CODE REVIEW

File: app/models/coordinator.rb [hunk 1]
```diff
@@ -1,17 +1,9 @@
...
+  belongs_to :affiliate
   belongs_to :user
 
-  has_many :enrollments
-  has_many :students, through: :enrollments
-
-  has_many :volunteer_jobs
-  has_many :tutors, through: :volunteer_jobs
-
-  has_many :affiliates
```
Comment (xingyif) [inline app/models/coordinator.rb:13]: if we delete this, how do we keep track of this coordinator's previous affiliates?

File: app/models/student.rb [hunk 2]
```diff
@@ -9,7 +9,7 @@ class Student < ApplicationRecord
   has_many :assessments
 
   has_many :enrollments
-  has_many :coordinators, through: :enrollments
```
Comment (xingyif) [inline app/models/student.rb:12]: iirc, roberta wants student to have coordinator, not affiliates

File: app/models/coordinator.rb [hunk 3]
```diff
@@ -1,17 +1,9 @@
 class Coordinator < ApplicationRecord
   VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
   VALID_PHONE_REGEX = /\A\([0-9]{3}\) [0-9]{3}-[0-9]{4}\z/
-
+  belongs_to :affiliate
```
Comment (xingyif) [inline app/models/coordinator.rb:4]: coordinator only keeps track of one affiliate?

File: spec/models/enrollment_spec.rb [hunk 4]
```diff
@@ -1,5 +1,13 @@
 require 'rails_helper'
 
 RSpec.describe Enrollment, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (xingyif) [inline spec/models/enrollment_spec.rb:4]: FINALLY! got rid of the pending spec :)

File: app/models/coordinator.rb (hunk 3 above)
Comment (zack) [reply app/models/coordinator.rb:4]: As far as I am aware this reflects reality.

File: app/models/coordinator.rb (hunk 1 above)
Comment (zack) [reply app/models/coordinator.rb:13]: If we need history then we need a third model, which we've never had.

File: app/models/student.rb (hunk 2 above)
Comment (zack) [reply app/models/student.rb:12]: A student can have a coordinator through an affiliate. But imagine that a coordinator is fired... Do you think it makes sense to go update 200 students to the new coordinator? No, that's silly.

File: app/models/coordinator.rb (hunk 3 above)
Comment (zack) [reply app/models/coordinator.rb:4]: Nobody is coordinating multiple affiliates.

File: app/models/student.rb (hunk 2 above)
Comment (zack) [reply app/models/student.rb:12]: It's like... the student registers at an affiliate and then the coordinator is in charge of that affiliate. The student doesn't register at or with a coordinator... The coordinator is coordinating the affiliate.

**Review by xingyif — APPROVED** [review]

**Pull request merged**